client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -118,23 +121,27 @@ const Checkout = React.createClass( {
 	},
 
 	addProductToCart: function() {
-		const planSlug = getUpgradePlanSlugFromPath( this.props.product, this.props.selectedSite );
-
-		let cartItem,
-			cartMeta;
-
-		if ( planSlug ) {
-			cartItem = getCartItemForPlan( planSlug );
+		const { product, productsList, purchaseId, selectedSite, selectedSiteSlug } = this.props;
+		const meta = product.split( ':' )[ 1 ];
+		const productSlug = getUpgradePlanSlugFromPath( product, selectedSite.jetpack ) || product.split( ':' )[ 0 ];
+
+		let cartItem = getCartItemForPlan( productSlug );
+
+		if ( 'theme' === productSlug ) {
```

Comment:
Here we're changing the logic for capturing various item types; I'd expect to see this as part of the testing instructions, otherwise we might be introducing subtle bugs with some cart item types. Also, would it make sense to add some tests prior to introducing these changes, so we can be more certain we're not breaking stuff?